api/src/main/scala/org/ensime/api/common.scala

```diff
@@ -93,8 +93,8 @@ object RefactorType {
 }
 
 final case class SourceFileInfo(
-    file: File,
-    contents: Option[String] = None,
+    file: EnsimeFile,
+    @deprecating("prefer contentsIn, contents will be removed") contents: Option[String] = None,
```

Comment:
oh, but you're not using wire protocols! Ah, well then you've got a legitimate use case for this and I'll remove the `@deprecating` and replace it with a comment explaining the intended uses. I still don't like that it is two `Option`s... it should really be a sealed family, otherwise what do we do when both are `Some`? :-/